src/test/java/org/folio/rest/api/ItemEffectiveCallNumberTest.java

```diff
@@ -110,6 +110,142 @@ public void canInitializeEffectiveCallNumberAfterItemChange() throws Exception {
     assertThat(getItem(item.getId()).getEffectiveCallNumberComponents().getCallNumber(), is("updatedCallNumber"));
   }
 
+  @Test
+  public void canInitializeEffectiveCallNumberPrefix() throws Exception {
+
+    UUID holding = createInstanceAndHoldingWithCallNumberPrefix(mainLibraryLocationId);
+    Item item = buildItem(holding, null, null);
+    EffectiveCallNumberComponents components = new EffectiveCallNumberComponents();
+    item.setEffectiveCallNumberComponents(components);
+
+    String query = String.format("INSERT INTO test_tenant_mod_inventory_storage.item (id, jsonb) values ('%s','%s');",
+        item.getId(), mapper.writeValueAsString(item));
+    runSql(query);
+
+    assertThat(getItem(item.getId()).getEffectiveCallNumberComponents().getPrefix(), is(nullValue()));
+
+    runSql(POPULATE_EFFECTIVE_CALL_NUMBER_SQL);
+
+    assertThat(getItem(item.getId()).getEffectiveCallNumberComponents().getPrefix(), is("testCallNumberPrefix"));
+  }
+
+  @Test
+  public void canInitializeEffectiveCallNumberPrefixAfterHoldingsChange() throws Exception {
+
+    UUID holding = createInstanceAndHoldingWithCallNumberPrefix(mainLibraryLocationId);
+    Item item = buildItem(holding, null, null);
+    createItem(item);
+
+    assertThat(getItem(item.getId()).getEffectiveCallNumberComponents().getPrefix(), is("testCallNumberPrefix"));
+
+    // Update holdings directly without updating item
+    String template = "UPDATE test_tenant_mod_inventory_storage.holdings_record SET jsonb = jsonb_set(jsonb, '{callNumberPrefix}', '\"%s\"') WHERE id = '%s';";
+    String query = String.format(template, "updatedCallNumberPrefix", holding.toString());
+    runSql(query);
+
+    assertThat(getHoldings(holding).getCallNumberPrefix(), is("updatedCallNumberPrefix"));
+    assertThat(getItem(item.getId()).getEffectiveCallNumberComponents().getPrefix(), is("testCallNumberPrefix"));
+
+    runSql(POPULATE_EFFECTIVE_CALL_NUMBER_SQL);
+
+    assertThat(getHoldings(holding).getCallNumberPrefix(), is("updatedCallNumberPrefix"));
+    assertThat(getItem(item.getId()).getEffectiveCallNumberComponents().getPrefix(), is("updatedCallNumberPrefix"));
+  }
+
+  @Test
+  public void canInitializeEffectiveCallNumberPrefixAfterItemChange() throws Exception {
```

Comment:
I'm not sure I understand this test. As I understand it, these are for testing the migration script. And that is a static operation, why is it relevant that an item cold change?

I think this is demonstrating that the initial value should be the item level prefix when it is present?

If so, I suggest simplifying this and renaming it to
```suggestion
  public void canInitializeEffectiveCallNumberPrefixToItemLevelWhenPresent() throws Exception {
```